### PR TITLE
enh(ui) update duplicate dialog to support an extra property: limit 

### PR DESCRIPTION
**Pull request by Med1233**
## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/packages/ui/src/Dialog/Duplicate/index.test.tsx [hunk 1]
```diff
@@ -45,4 +45,47 @@ describe('DialogDuplicate', () => {
...
+        onConfirm={mockConfirm}
+      />
+    );
+
+    const input = getByDisplayValue('1');
+    fireEvent.change(input, { target: { value: '' } });
+
+    expect(getByText('confirm')).toBeDisabled();
+  });
+  it('disactivates the confirm button if the given number is more than the limit', () => {
```
Comment (Thebarda) [inline centreon/packages/ui/src/Dialog/Duplicate/index.test.tsx:68]:
```suggestion

  it('disables the confirm button if the given number is more than the limit', () => {
```

File: centreon/packages/ui/src/Dialog/Duplicate/index.tsx [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+  container: {
+    width: theme.spacing(30)
+  }
+}));
+
 const Duplicate = ({
   labelInput = 'Count',
   labelConfirm = 'Duplicate',
   labelTitle = 'Duplicate',
+  limit = Infinity,
```
Comment (Thebarda) [inline centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: Does the undefined value work?

Comment (Thebarda) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: You should take into account that the limit might ne undefined then

Comment (Med1233) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: if the limit props were not given then the value is the default one: Infinity

Comment (Med1233) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: and I have verified this in the other stories

Comment (Thebarda) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: Or you can a little bit of code to handle undefined

Comment (Med1233) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: I don't understand! I already did this. Is it a bad practice to use a default value ?

Comment (Thebarda) [reply centreon/packages/ui/src/Dialog/Duplicate/index.tsx:25]: Both are valid. We can keep this way

**Review by Thebarda — APPROVED** [review]

**Review by Yassir-BenBOUBKER — APPROVED** [review]

**Pull request merged**